lib/authz/permissions.go

```diff
@@ -1041,8 +1041,9 @@ func definitionForBuiltinRole(clusterName string, recConfig types.SessionRecordi
 						types.NewRule(types.KindOktaAssignment, services.RW()),
 						types.NewRule(types.KindProxy, services.RO()),
 						types.NewRule(types.KindClusterAuthPreference, services.RO()),
-						types.NewRule(types.KindRole, services.RO()),
+						types.NewRule(types.KindRole, services.RW()),
```

Comment:
Are there any security concerns about this? there were some discussions when granting the RW for `types.KindUser` https://github.com/gravitational/teleport/pull/33630

Reply:
@jentfoo do you have any thoughts? We kind of need to be able to create roles and access lists as part of this.

Reply:
@mdwn in #33630 we discussed controls so that Okta was only managing users created within Okta.  My initial concern around Access Lists is that this would provide a bypass of that control (as non-Okta users could be added to the Access List).

How can we ensure that Okta is only able to manage Okta managed users and not users created outside of Okta?

Reply:
This should be doable via RBAC, but let me double check if we have the bits and pieces we require to make that happen. If not, I'll add them.

Reply:
Reviewing the most recent commit, it seems like we also need to add a control on the Access List side as well.  What would prevent manual additions of users to Okta managed lists as a means to allow Okta to manage users it should not?

Reply:
I've updated the PR to handle this, but I'm still testing this out.